handler.go

```diff
@@ -78,31 +80,37 @@ func (a *podInjector) Handle(ctx context.Context, req admission.Request) admissi
 	if skip(pod) {
 		return admission.Allowed("telegraf-injector has no power over this pod")
 	}
-	classData, err := a.getClassData(pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
+
 	name := pod.GetName()
 	if name == "" {
 		name = names.SimpleNameGenerator.GenerateName(pod.GetGenerateName())
 		pod.SetName(name)
 		handlerLog.Info("name: " + name + ",  pod_getname=" + pod.GetName())
 	}
-	secret, err := addSidecar(pod, pod.GetName(), req.Namespace, classData)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-	if req.Operation == admv1.Create {
-		err = a.client.Create(ctx, secret)
+
+	classData, err := a.getClassData(pod)
+	if err == nil {
```

Comment:
This wouldn't work because if err != nil, then you into the `else` statement, it logs the error but then continue without the right informations. I like to fail fast. I think the right approach if err != nil is from L81 (the skip function) because that's what we actually want to do

Reply:
You are right, I can just return empty `Allowed` response. Very good point.

FYI this worked and logged message correctly, but I agree, with the new approach I just pushed it is much cleaner.